packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/scheduleWalkthrough.ts

```diff
@@ -0,0 +1,170 @@
+          scheduleParams.cloudwatchRule = 'NONE';
+          delete cfnContent.Resources.CloudWatchEvent;
+          delete cfnContent.Resources.PermissionForEventsToInvokeLambda;
+          delete cfnContent.Outputs.CloudWatchEventRule;
+          break;
+        }
+      }
+      fs.writeFileSync(cfnFilePath, JSON.stringify(cfnContent, null, 4));
+    }
+  }
+  return scheduleParams;
+}
+
+async function cronServiceWalkthrough(context: any) {
+  let cloudwatchRule;
+  // resource questions for setting cron
+  const intervalQuestion = {
+    type: 'list',
+    name: 'interval',
+    message: 'At which interval should the function be invoked:',
+    choices: [MINUTES, HOURLY, DAILY, WEEKLY, MONTHLY, YEARLY, CUSTOM],
+  };
+  const intervalAnswer = await inquirer.prompt([intervalQuestion]);
+  switch (intervalAnswer.interval) {
+    case MINUTES: {
+      cloudwatchRule = minuteHelper(context);
+      break;
+    }
+    case HOURLY: {
+      cloudwatchRule = hourHelper(context);
+      break;
+    }
+    case DAILY: {
+      var exp = new CronBuilder();
+      exp = await timeHelper(exp);
+      cloudwatchRule = exp.build();
+      cloudwatchRule = 'cron(' + replaceAt(cloudwatchRule, cloudwatchRule.lastIndexOf('*'), '?') + ' ' + '*' + ')';
+      break;
+    }
+    case WEEKLY: {
+      var exp1 = new CronBuilder();
+      exp1 = await weekHelper(exp1);
+      exp1 = await timeHelper(exp1);
+      cloudwatchRule = exp1.build();
+      cloudwatchRule = 'cron(' + cloudwatchRule + ' ' + '*' + ')';
+      break;
+    }
+    case MONTHLY: {
+      var exp2 = new CronBuilder();
+      exp2 = await monthHelper(exp2, context);
+      exp2 = await timeHelper(exp2);
+      cloudwatchRule = exp2.build();
+      cloudwatchRule = 'cron(' + replaceAt(cloudwatchRule, cloudwatchRule.lastIndexOf('*'), '?') + ' ' + '*' + ')';
+      break;
+    }
+    case YEARLY: {
+      var exp3 = new CronBuilder();
+      exp3 = await yearHelper(exp3, context);
+      exp3 = await timeHelper(exp3);
+      cloudwatchRule = exp3.build();
+      cloudwatchRule = 'cron(' + replaceAt(cloudwatchRule, cloudwatchRule.lastIndexOf('*'), '?') + ' ' + '*' + ')';
+
+      break;
+    }
+    case CUSTOM: {
+      const customRuleQuestion = {
+        type: 'input',
+        name: 'customRule',
+        message: 'Custom Schedule expression(Learn more: https://amzn.to/3akXtJF)',
+        validate: ValidCronExpression({
+          onErrorMsg: 'Enter a valid Schedule Expression (Learn more: https://amzn.to/3akXtJF)',
+        }),
+      };
+      const customRuleAnswer = await inquirer.prompt([customRuleQuestion]);
+      cloudwatchRule = 'cron(' + customRuleAnswer.customRule + ')';
+
+      break;
+    }
+  }
+  return cloudwatchRule;
+}
+
+function replaceAt(string, index, replace) {
+  return string.substring(0, index) + replace + string.substring(index + 1);
+}
+
+module.exports = {
+  scheduleWalkthrough,
+  cronServiceWalkthrough,
+  isValidCronExpression,
+};
```

Comment:
not necessary with TS

Reply:
yes, just export the functions straight ahead